packages/daemon/DESIGN.md

```diff
@@ -0,0 +1,90 @@
+
+Endo is a user agent for running guest programs.
+These programs receive capabilities from the user in the form of references to
+potentially powerful remote objects.
+To that end, Endo exhibits a system called Eventual Send that stretches
+references to remote objects over one or more capability transfer protocols
+(CapTP).
+
+A Promise represents a result from either the past or the future, and Eventual
+Send enables programs to interact with those eventual results, even if they are
+in another worker or on the far side of a network.
+Furthermore, Eventual Send allows messages to be pipelined to these eventual
+results to overcome any latency that might otherwise accrue if the program
```

Comment:
FYI just noticing word choices that jump out to me as odd. "accrue". No change suggested.

Reply:
Please link to something on promise pipelining, since you don't (and should not) explain it in this document. Some suggestions:

http://erights.org/elib/distrib/pipeline.html
https://capnproto.org/rpc.html
https://en.wikipedia.org/wiki/Futures_and_promises#Promise_pipelining
https://www.proposals.es/proposals/Support%20for%20Distributed%20Promise%20Pipelining